hs_core/hs_rdf.py

```diff
@@ -0,0 +1,358 @@
+        for generic_relation in generic_relations:
+            generic_relation.related_model.ingest_rdf(graph, subject, self)
+        self.save()
+
+    def get_rdf_graph(self):
+        """adds the rdf triples of all generic relations on the object into an rdflib Graph"""
+        graph = Graph()
+        graph.namespace_manager = NAMESPACE_MANAGER
+
+        subject = self.rdf_subject()
+        # graph.add((subject, RDF.type, self.rdf_type()))
+        generic_relations = list(filter(lambda f: isinstance(f, GenericRelation), type(self)._meta.virtual_fields))
+        for generic_relation in generic_relations:
+            for f in getattr(self, getattr(generic_relation, 'name', None), None).all():
+                f.rdf_triples(subject, graph)
+
+        return graph
+
+    def get_xml(self, pretty_print=True, include_format_elements=True):
+        """Generates ORI+RDF xml for this metadata"""
+        # ORE = Namespace("http://www.openarchives.org/ore/terms/")
+        g = self.get_rdf_graph()
+        # g.add((self.rdf_metadata_subject(), ORE.describes, self.rdf_subject()))
+        return g.serialize(format='hydro-xml').decode()
+
+
+class RDF_Term_MixIn(object):
+    """
+    Provides methods for serializing a django model into and rdflib triples and deserializing from an rdflib Graph
+     back to a django model.  This mixin is designed for django models that are generic relations of another django
+     model which represents a group of metadata.
+
+     Without any configuration, the rdf_triples(subject) method will add all fields on the generic relation model except
+     for 'id', 'object_id', 'content_type' will be serialized as rdf triples.  If rdf_term is defined, it will be
+     used as the predicate.  If rdf_term is not defined, one will be defined using the class name and the
+     HSTERMS namespace.
+
+     When a model requires an rdf-xml format that the default configuration will not provide, you may overwrite
+     the rdf_triples method and the ingest_rdf method.  If you overwrite one, you must overwrite both or this mixin
+     will not work correctly.
+     """
+
+    ignored_fields = ['id', 'object_id', 'content_type']
+
+    def rdf_triples(self, subject, graph):
+        """Default implementation that parses by convention."""
+        term = self.get_class_term()
+        if not term:
+            metadata_node = subject
+        else:
+            metadata_node = BNode()
+            graph.add((subject, term, metadata_node))
+        for field_term, field_value in self.get_field_terms_and_values():
+            graph.add((metadata_node, field_term, field_value))
+
+    def get_field_terms_and_values(self, extra_ignored_fields=[]):
+        """Method that returns the field terms and field values on an object"""
+        term_values = []
+        extra_ignored_fields.extend(self.ignored_fields)
+        for field in self._meta.fields:
+            if field.name in extra_ignored_fields:
+                continue
```

Comment:
continue statement with the check can probably be avoided by using list filter to filter out ignored_fields from self._meta.fields first.

Reply:
So, something like this?
`for field in [field for field in self._meta.fields if field.name not in extra_ignored_fields]:`